plugin/src/test/java/hudson/plugins/swarm/SwarmClientIntegrationTest.java

```diff
@@ -148,6 +163,83 @@ public void addRemoveLabelsViaFileWithoutUniqueIdLong() throws Exception {
         addRemoveLabelsViaFile(labelsToRemove, labelsToAdd, false);
     }
 
+
+    @Test
+    public void pidFilePreventsStart() throws Exception {
+        // Start the first client with a pid file and ensure it's up.
+        Node node1 = TestUtils.createSwarmClient(
+                j,
+                processDestroyer,
+                temporaryFolder,
+                "-pidFile",
+                getPidFile().getAbsolutePath());
+
+        // Try to start a second and ensure it fails to run.
+        // Do not wait for it to come up on the server side.
+        TestUtils.SwarmClientProcessWrapper node2 = TestUtils.runSwarmClient(
+                "agent_fail",
+                j,
+                processDestroyer,
+                temporaryFolder,
+                "-pidFile",
+                getPidFile().getAbsolutePath());
+
+        node2.process.waitFor(5, TimeUnit.SECONDS);
+        assertFalse("Second client should fail to start", node2.process.isAlive());
+        assertEquals("Exit code", 1, node2.process.exitValue());
+        assertTrue("Log message mentions 'already exists'",
+                Files.readAllLines(node2.stderr.toPath()).stream().anyMatch(line -> line.contains("already exists")));
+    }
+
+    @Test
+    public void pidFileForStaleProcessIsIgnored() throws Exception {
+
+        Files.write(getPidFile().toPath(), "66000".getBytes());
+
+        // Pid file should be ignored since the process isn't running.
+        TestUtils.SwarmClientProcessWrapper node = TestUtils.runSwarmClient(
+                "agent0",
+                j,
+                processDestroyer,
+                temporaryFolder,
+                "-pidFile",
+                getPidFile().getAbsolutePath());
+
+        // ensure that the client starts.
+        TestUtils.waitForNode("agent0", j);
```

Comment:
[nit] Should we combine the calls to `TestUtils.runSwarmClient` and `TestUtils.waitForNode` into a single call to `TestUtils.createSwarmClient`?

Reply:
This is what happens when I do more ops than dev... I'm impressed that you noticed all this.